vitrivr-engine-core/src/main/kotlin/org/vitrivr/engine/core/model/types/Type.kt

```diff
@@ -1,20 +1,171 @@
 package org.vitrivr.engine.core.model.types
 
+import java.util.*
+
 /**
- * An enumeration of supported [Type]s
+ * The [Type]s supported by vitrivr-engine.
  *
  * @author Ralph Gasser
  * @author Loris Sauter
- * @version 1.1.0
+ * @version 2.0.0
  */
-enum class Type {
-    STRING,
-    BOOLEAN,
-    BYTE,
-    SHORT,
-    INT,
-    LONG,
-    FLOAT,
-    DOUBLE,
-    DATETIME
+sealed interface Type {
+
+    companion object {
+        /**
+         * Returns the [Type] for the given [String] representation.
+         *
+         * @param type [String] representation of the [Type]
+         * @return [Type]
+         */
+        fun valueOf(type: kotlin.String, dimensions: kotlin.Int = 0): Type {
+            return when (type.uppercase()) {
+                "TEXT" -> Text
+                "STRING" -> String
+                "BOOLEAN" -> Boolean
+                "BYTE" -> Byte
+                "SHORT" -> Short
+                "INT" -> Int
+                "LONG" -> Long
+                "FLOAT" -> Float
+                "DOUBLE" -> Double
+                "DATETIME" -> Datetime
+                "BOOLEANVECTOR" -> BooleanVector(dimensions)
+                "INTVECTOR" -> IntVector(dimensions)
+                "LONGVECTOR" -> LongVector(dimensions)
+                "FLOATVECTOR" -> FloatVector(dimensions)
+                "DOUBLEVECTOR" -> DoubleVector(dimensions)
+                else -> throw IllegalArgumentException("Type $type is not supported!")
+            }
+        }
+    }
+
+
+    /** The number of dimensions for this [Type]. */
+    val dimensions: kotlin.Int
```

Comment:
You could just initialize this with 1 and only override the ones where the dimensionality is actually type-dependent, no?